src/test/util/random.cpp

```diff
@@ -7,32 +7,39 @@
 #include <logging.h>
 #include <random.h>
 #include <uint256.h>
+#include <util/check.h>
 
 #include <cstdlib>
-#include <string>
+#include <iostream>
 
 FastRandomContext g_insecure_rand_ctx;
 
 extern void MakeRandDeterministicDANGEROUS(const uint256& seed) noexcept;
 
 void SeedRandomForTest(SeedRand seedtype)
 {
-    static const std::string RANDOM_CTX_SEED{"RANDOM_CTX_SEED"};
+    constexpr auto RANDOM_CTX_SEED{"RANDOM_CTX_SEED"};
 
     // Do this once, on the first call, regardless of seedtype, because once
     // MakeRandDeterministicDANGEROUS is called, the output of GetRandHash is
     // no longer truly random. It should be enough to get the seed once for the
     // process.
     static const uint256 ctx_seed = []() {
         // If RANDOM_CTX_SEED is set, use that as seed.
-        const char* num = std::getenv(RANDOM_CTX_SEED.c_str());
-        if (num) return uint256S(num);
+        if (const char* num{std::getenv(RANDOM_CTX_SEED)}) {
+            if (auto num_parsed{uint256::FromHex(num)}) {
```

Comment:
re: https://github.com/bitcoin/bitcoin/pull/30569#discussion_r1712563888

Thanks for the links. I don't think I realized this change was made in the other PR, and I think it is not a good change.

I don't see a reason somebody shouldn't be able to run `RANDOM_CTX_SEED=12345 ./test_bitcoin` conveniently from a command line or script without padding the seed with 59 0's. It seems like this makes it pointlessly difficult to specify a random seed and I don't get the motivation behind it.

If we want to go ahead with change, I think it should be described accurately as making something that used to work no longer work, not just as adding error checking and implying that anything this now rejects was previously in error.

Reply:
> It seems like this makes it pointlessly difficult to specify a random seed and I don't get the motivation behind it.

I definitely didn't mean to introduce user friction with this PR - the philosophy was more to not introduce unnecessary code complexity for features that aren't used, but since it seems like this might be breaking existing/valuable use cases, I've now updated `RANDOM_CTX_SEED` to use the more lenient `uint256::FromUserHex()` to still allow 0x-prefixes and shorter input, and it doesn't add code complexity anyway.